allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -22,36 +24,98 @@ class PretrainedTransformerTokenizer(Tokenizer):
     """
 
     def __init__(
-        self, model_name: str, start_tokens: List[str] = None, end_tokens: List[str] = None
+        self,
+        model_name: str,
+        start_tokens: List[str] = None,
+        end_tokens: List[str] = None,
+        add_special_tokens: bool = True,
+        max_length: int = None,
+        stride: int = 0,
+        truncetion_strategy: str = "longest_first",
     ) -> None:
         self._tokenizer = AutoTokenizer.from_pretrained(model_name)
-        default_start_tokens, default_end_tokens = _guess_start_and_end_token_defaults(model_name)
-        self._start_tokens = start_tokens if start_tokens is not None else default_start_tokens
-        self._end_tokens = end_tokens if end_tokens is not None else default_end_tokens
+        self._start_tokens = start_tokens
+        self._end_tokens = end_tokens
+        self._add_special_tokens = add_special_tokens
+        self._max_length = max_length
+        self._stride = stride
+        self._truncation_strategy = truncetion_strategy
+
+    def _add_custom_start_end_tokens(self, text: str) -> str:
+        if self._start_tokens is not None:
+            text = " ".join(self._start_tokens) + " " + text
+        if self._end_tokens is not None:
+            text = text + " " + " ".join(self._end_tokens)
+        return text
 
     @overrides
     def tokenize(self, text: str) -> List[Token]:
+        """
+        This can also handle sentence pairs. In this case 1st and 2nd sentences in ``text``
+        are expected to be separated with ``ALLENNLP_SENTENCE_PAIR_SEP`` symbol.
+        """
         # TODO(mattg): track character offsets.  Might be too challenging to do it here, given that
         # ``transformers``` is dealing with the whitespace...
-        token_strings = self._start_tokens + self._tokenizer.tokenize(text) + self._end_tokens
-        return [Token(t) for t in token_strings]
+        if ALLENNLP_SENTENCE_PAIR_SEP in text:
+            sentence_1, sentence_2 = text.split(ALLENNLP_SENTENCE_PAIR_SEP)
+            sentence_1 = self._add_custom_start_end_tokens(sentence_1)
+            sentence_2 = self._add_custom_start_end_tokens(sentence_2)
+        else:
+            sentence_1 = text
+            sentence_1 = self._add_custom_start_end_tokens(sentence_1)
+            sentence_2 = None
+
+        encoded_tokens = self._tokenizer.encode_plus(
+            text=sentence_1,
+            text_pair=sentence_2,
+            add_special_tokens=self._add_special_tokens,
+            max_length=self._max_length,
+            stride=self._stride,
+            truncation_strategy=self._truncation_strategy,
+            return_tensors=None,
+        )
+        # token_ids containes final list with ids for both regualr and special tokens
+        token_ids = encoded_tokens["input_ids"]
+        token_type_ids = encoded_tokens["token_type_ids"]
 
+        token_strings = self._tokenizer.convert_ids_to_tokens(token_ids, skip_special_tokens=False)
```

Comment:
Seems like this is an extra iteration over the tokens that we don't need, given the iteration you're doing below.  Why not remove this line and do a simple vocab lookup inside the loop?